### DISCUSSION TITLE
Welcome to vavr Discussions!

### DISCUSSION THREAD

**Discussion by pivovarit** [vavr-io/vavr#2782] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by forresthopkinsa** [comment]
Hello sir! Excited to see some fresh faces in the maintenance of this critical library, cheers to you for taking up the torch

**Reply by pivovarit** [reply]
Thanks! It will take me some time to get up to speed, but I won't let this project degrade and die.

**Reply by dbercier** [reply]
Thank you for this quick response. This is definitely encouraging. I really want to become a contributor to the lib because it means that much to me. Please tell me what I have to do to do that?

**Reply by pivovarit** [reply]
Thanks! I think it's pretty straightforward: either have a look at the issues page and find something for yourself, or feel free to suggest something yourself. Just make sure to discuss the changes upfront to avoid doing work which will be rejected later on

**Reply by dbercier** [reply]
Thank you. Question, are all the issues listed approved for development? I wouldn't want to take something and then find out the functionality is green-lit

**Reply by pivovarit** [reply]
That's exactly the point - the state of issues is "all around the place", so "contributing" does not only mean "producing code", but also refining the issue and coming up with proposals. If you're unsure, always ask first

Having said that, I have just added `ready-to-implement` label that will be used for features that can be pretty much taken of the shelf and implemented right away

**Reply by dbercier** [reply]
Roger that. Thank you so much for this help. I will look for that label and pick up an issue and help keep VAVR moving forward.

**Reply by dbercier** [reply]
Do you have a specific, targeted release date for 0.10.5?

**Reply by pivovarit** [reply]
Should happen this weekend. The tag is already in place, but I'm not sure if the release infrastructure will stand the test of time

**Reply by dbercier** [reply]
Roger that. Sounds awesome!

**Reply by pivovarit** [reply]
https://github.com/vavr-io/vavr/releases/tag/v0.10.5 it's live :)

**Comment by matozoid** [comment]
Good, some activity here! I'm doing a personal project, a diagram editor, where I keep the code sane by using vavr. I can easily track edits in the diagram, so "undo" and "history" are free. When everything is mutable, that would become a nightmare quickly!

**Reply by matozoid** [reply]
@pivovarit would you like some tips to get going? I maintained this for five years: https://github.com/javaparser/javaparser

**Comment by AdrianLagartera** [comment]
Hi since a good mentor took me to collaborate in OSS projects I have been looking for projects to collaborate in, and since I saw this project in one of the posts in the Java reddit I would like to contribute as much as I can in this one, so it is a pleasure to see all the comments talking so well about it, I hope I can contribute something with my little knowledge, hahaha

**Comment by bkern** [comment]
Excited to see this library continue on. Long time user mostly for functional error handling. Interested in contributing.

**Comment by gualtierotesta** [comment]
@pivovarit Hi and thanks for your effort to keep this project maintained and alive.
I'm interested in contributing, especially on testing and documentation. Let me know where I could help.

**Reply by Andras-Csanyi** [reply]
The same from my side.

**Comment by dbercier** [comment]
Hello, I am an avid user of the VAVR library. I am working for a client that did allow us to use it but recently the lead architect is telling folks to remove it because of the CVE's being raised against it and because from their research they discovered that the 1.0 version of VAVR was abandoned because of breaking changes and complaints from the community. This is heartbreaking for me as an FP enthusiast. It's literally like my world is crashing not just while working at this client but anywhere in the future. Can anyone tell me the current state of things with VAVR and offer any hope for its future?

**Reply by pivovarit** [reply]
Hi! Let me clarify a few things:
- `1.0` was not _abandoned_ - some work that went into it will probably go down the drain, but `1.0` will happen at some point in the future. Yes, it will most likely contain breaking changes since bumping up a major version is the only opportunity to do so
- `0.x` will be actively maintained, actually this was posted literally an hour ago: https://x.com/vavr_io/status/1844678930034282511 
- which CVEs? there was just one and not applicable to production code: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15250 and it will be addressed in `0.10.5`
